tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -350,26 +353,13 @@ def assertXdsConfigExists(self, test_client: XdsTestClient):
                      json_format.MessageToJson(config, indent=2))
         self.assertSameElements(want, seen)
 
-    @staticmethod
-    def getRouteConfigVersion(test_client: XdsTestClient) -> Optional[str]:
-        config = test_client.csds.fetch_client_status(log_level=logging.INFO)
-        route_config_version = None
-        for xds_config in config.xds_config:
-            if xds_config.WhichOneof('per_xds_config') == "route_config":
-                route_config = xds_config.route_config
-                logger.info('Route config found: %s',
-                            json_format.MessageToJson(route_config, indent=2))
-                route_config_version = route_config.dynamic_route_configs[
-                    0].version_info
-                logger.info('found routing config version: %s',
-                            route_config_version)
-                break
-        return route_config_version
-
     def assertRouteConfigUpdateTrafficHandoff(
             self, test_client: XdsTestClient,
             previous_route_config_version: str, retry_wait_second: int,
             timeout_second: int):
+        logger.info(
+            'comparing route config versions, previous_route_config_version: %s',
+            previous_route_config_version)
```

Comment:
I would recommend to add a `config = None` here, in case the client config does not exist (I had a typo in comment last week).

Reply:
oh it may not be completely the same, but I learned a while ago that surfacing the exception might be good as well.
Plus we actually handle the situation by itself.
```
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/zivy/src/grpc/tools/run_tests/xds_k8s_test_driver/tests/api_listener_test.py", line 90, in test_api_listener
    json_format.MessageToDict(raw_config))
  File "/Users/zivy/src/grpc/tools/run_tests/xds_k8s_test_driver/venv/lib/python3.6/site-packages/google/protobuf/json_format.py", line 165, in MessageToDict
    return printer._MessageToJsonObject(message)
  File "/Users/zivy/src/grpc/tools/run_tests/xds_k8s_test_driver/venv/lib/python3.6/site-packages/google/protobuf/json_format.py", line 199, in _MessageToJsonObject
    message_descriptor = message.DESCRIPTOR
AttributeError: 'NoneType' object has no attribute 'DESCRIPTOR'

----------------------------------------------------------------------
Ran 1 test in 465.016s
```

Reply:
What I hope for is a clear log line when the client didn't return any xDS config. Surfacing the exception can be done via "raise", but we can help oncalls with an info log.

Reply:
> What I hope for is a clear log line when the client didn't return any xDS config. Surfacing the exception can be done via "raise", but we can help oncalls with an info log.

Thank you. I'm actually leaning towards what you are suggesting, but still, i guess it is minor.